base/set.jl

```diff
@@ -427,6 +427,18 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
+function allunique(x::AbstractArray)
+    if length(x) < 32  # then linear search is faster, even in worst case
+        return @inbounds all(!(x[i] in @view x[i+1:end]) for i in LinearIndices(x))
```

Comment:
It's possible this ought to be restricted to AbstractVector, or DenseArray, as linear indexing is slow on a transposed matrix:
```
julia> for n in 5:5:20
         @show 2n
         x = rand(2,n)'  # Does not support fast linear indexing
         @btime allunique($x)
         @btime _allunique($x) # this PR
       end
2n = 10
  min 120.645 ns, mean 134.354 ns (4 allocations, 400 bytes)
  min 154.554 ns, mean 155.407 ns (0 allocations)
2n = 20
  min 262.725 ns, mean 293.887 ns (7 allocations, 1.12 KiB)
  min 455.808 ns, mean 457.937 ns (0 allocations)
2n = 30
  min 319.447 ns, mean 351.454 ns (7 allocations, 1.12 KiB)
  min 917.857 ns, mean 927.596 ns (0 allocations)
2n = 40
  min 633.335 ns, mean 1.013 μs (10 allocations, 3.62 KiB)
  min 1.479 μs, mean 1.509 μs (0 allocations)
```